papermill/execute.py

```diff
@@ -79,7 +77,6 @@ def preprocess(self, nb, resources):
 
         for index, cell in execution_iterator:
             cell.metadata["papermill"]["status"] = RUNNING
-            future = executor.submit(write_ipynb, nb, output_path)
```

Comment:
this change looks wrong; you sure you meant to mess with the future pool execution?

Reply:
@MSeal Thanks for your comment, here if I don't use output option in https://github.com/nteract/papermill/blob/master/papermill/execute.py#L138, I will get output file error, if I remove these code, it would be OK, perhaps I need a better way to deal with this.

Reply:
You need to check if `output_path is None` and not always write.

Reply:
@betatim Thanks, I added this now. :)